### DISCUSSION TITLE
Enhancement: Toggle for infinite scroll

### DISCUSSION THREAD

**Discussion by Fireye04** [bluesky-social/atproto#3233] [Bluesky Client App]
Hi! Not sure if this is the right spot for this, but I'd love to see a way to optionally toggle off infinite scrolling on content feeds. Toggling off infinite scrolling would reintroduce more stimuli for users to end sessions, improving mindful/ healthy use of the application. For users who are conscious about the amount of their time spent on social media, or those who are looking to reduce it, an option to toggle off infinite scrolling would be incredibly beneficial.

Just wanted to float the idea and open up discussion. Have a great day, and thanks for the awesome app.

**Comment by himham-jak** [comment]
I like this idea. It feels very "user-owned experience".

Small question for you, what would it toggle to? A traditional page 1/100 setup? Items per page?

**Reply by Fireye04** [reply]
Could have a 100 post default, with a dropdown with a couple other options? 

Initial implementation could just be that 100 post toggle if the dropdown is too much.

**Comment by erwi0014** [comment]
I came on here to suggest this very feature! We need to bring back pages!!

**Comment by avleen** [comment]
Hi folks, a couple of us are working on implementing this feature.
We've got it working on the main feed and searches - both the search results and the `Discover feeds` scroll.

The app has a hard coded minimum posts to fetch limit of 30 posts, so we started with 30 as the default but made it slightly tunable.

Is this something folks at Bluesky would be interesting in having?
Who could we talk to, to see if there's interest?

Thanks!

<img width="1079" height="1295" alt="image" src="https://github.com/user-attachments/assets/f34bcef8-dc5d-4ec4-924c-13f559c3928f" />
<img width="1079" height="823" alt="image" src="https://github.com/user-attachments/assets/c3ba0af4-3ae1-4478-a33a-dee101c850c6" />
<img width="1080" height="532" alt="image" src="https://github.com/user-attachments/assets/b8f71db7-06c0-4baf-a9f9-e4fe374a96ee" />
